gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/IdentityProviderServiceProxyImpl.java

```diff
@@ -0,0 +1,170 @@
+                .flatMap(oldIdP -> updateSensitiveData(updateIdentityProvider, oldIdP))
+                .flatMap(updateIdentityProvider1 -> identityProviderService.update(domain, id, updateIdentityProvider1, principal));
+    }
+
+    @Override
+    public Completable delete(ReferenceType referenceType, String referenceId, String identityProviderId, User principal) {
+        return identityProviderService.delete(referenceType, referenceId, identityProviderId, principal);
+    }
+
+    @Override
+    public Completable delete(String domain, String identityProviderId, User principal) {
+        return identityProviderService.delete(domain, identityProviderId, principal);
+    }
+
+    private Single<IdentityProvider> filterSensitiveData(IdentityProvider idp) {
+        return identityProviderPluginService.getSchema(idp.getType())
+                .switchIfEmpty(Single.error(new IdentityProviderPluginSchemaNotFoundException(idp.getType())))
+                .map(schema -> {
+                    JsonNode configurationNode = objectMapper.readTree(idp.getConfiguration());
+                    JsonNode schemaNode = objectMapper.readTree(schema);
+                    if (schemaNode.has("properties")) {
+                        schemaNode.get("properties").fields().forEachRemaining(entry -> {
+                            if (entry.getValue().has("sensitive") &&
+                                    entry.getValue().get("sensitive").asBoolean()) {
+                                ((ObjectNode) configurationNode).put(entry.getKey(), SENSITIVE_VALUE);
+                            }
+                        });
+                        idp.setConfiguration(configurationNode.toString());
+                    }
+                    return idp;
+                });
+    }
+
+    private Single<UpdateIdentityProvider> updateSensitiveData(UpdateIdentityProvider updateIdentityProvider, IdentityProvider oldIdentityProvider) {
+        return identityProviderPluginService.getSchema(oldIdentityProvider.getType())
+                .switchIfEmpty(Single.error(new IdentityProviderPluginSchemaNotFoundException(oldIdentityProvider.getType())))
+                .map(schema -> {
+                    JsonNode updatedConfigurationNode = objectMapper.readTree(updateIdentityProvider.getConfiguration());
+                    JsonNode oldConfigurationNode = objectMapper.readTree(oldIdentityProvider.getConfiguration());
+                    JsonNode schemaNode = objectMapper.readTree(schema);
+                    if (schemaNode.has("properties")) {
+                        schemaNode.get("properties").fields().forEachRemaining(entry -> {
+                            if (entry.getValue().has("sensitive") &&
+                                    entry.getValue().get("sensitive").asBoolean()) {
+                                if (SENSITIVE_VALUE.equals(updatedConfigurationNode.get(entry.getKey()).asText())) {
```

Comment:
Hi, in order to limit the risk of overwriting in case of false manipulation (deletion of a * for example in the ui) is it possible at the front end to make the displayed value of the field in the form not be the real value (null value?). And check here that the value is null or only space maybe.